systemd/systemd.go

```diff
@@ -1392,22 +1401,39 @@ const (
 	snappyOriginModule = "X-SnapdOrigin"
 )
 
-func writeMountUnitFile(u *MountUnitOptions) (mountUnitName string, err error) {
+func ensureMountUnitFile(u *MountUnitOptions) (mountUnitName string, modified mountUpdateStatus, err error) {
 	if u == nil {
-		return "", errors.New("writeMountUnitFile() expects valid mount options")
+		return "", mountUnchanged, errors.New("ensureMountUnitFile() expects valid mount options")
 	}
 
 	mu := MountUnitPathWithLifetime(u.Lifetime, u.Where)
-	outf, err := osutil.NewAtomicFile(mu, 0644, 0, osutil.NoChown, osutil.NoChown)
+	tmpf, err := ioutil.TempFile("", filepath.Base(mu))
+	defer os.Remove(tmpf.Name())
 	if err != nil {
-		return "", fmt.Errorf("cannot open mount unit file: %v", err)
+		return "", mountUnchanged, fmt.Errorf("cannot open temporary mount unit file: %v", err)
 	}
-	defer outf.Cancel()
 
-	if err := parsedMountUnitTemplate.Execute(outf, &u); err != nil {
-		return "", fmt.Errorf("cannot generate mount unit: %v", err)
+	if err := parsedMountUnitTemplate.Execute(tmpf, &u); err != nil {
+		return "", mountUnchanged, fmt.Errorf("cannot generate mount unit: %v", err)
 	}
-	return filepath.Base(mu), outf.Commit()
+
+	if osutil.FileExists(mu) {
+		modified = mountUpdated
+	} else {
+		modified = mountCreated
+	}
+	stateErr := osutil.EnsureFileState(mu, &osutil.FileReferencePlusMode{
```

Comment:
instead of a TempFile this now could use a bytes.Buffer and a osutil.MemoryFileState, no?

Reply:
`template.Template.Execute` expects a file. I did not find anything in the standard go library or in osutil that provides a writable memory file. Is there anything I can use?

Reply:
I meant an `io.Writer`.

Reply:
I'm confused by this.  bytes.Buffer is a io.Writer and you can find other examples in the code base using template and bytes.Buffer together, see wrappers/services.go for example

Reply:
Done.